manifests/augeas.pp

```diff
@@ -42,35 +42,36 @@
   $plugin           = false,
   $restart          = false,
 ) {
-
-  include ::jenkins
-  include ::jenkins::cli
-
   validate_string($config_filename)
+  if ! is_string($changes) and ! is_array($changes) {
+    fail('$changes must be string or array.')
+  }
+  if ! is_string($onlyif) and ! is_array($onlyif) {
+    fail('$onlyif must be string or array.')
+  }
   validate_string($context)
+  validate_string($plugin_version)
+  if ! is_bool($plugin) and ! is_string($plugin) {
+    fail('$plugin must be bool or string.')
+  }
+  validate_bool($restart)
 
-  # validate $plugin embedded in case.
-
-  if $plugin_version { validate_string($plugin_version) }
+  include ::jenkins
+  include ::jenkins::cli
 
-  # changes and onlyif can be both a string or an array...
-  if ! is_string($onlyif) { validate_array($onlyif) }
-  if ! is_string($changes) { validate_array($changes) }
 
   case $plugin {
-    true,'true': { # lint:ignore:quoted_booleans
+    true: {
       jenkins::plugin {$name:
         version       => $plugin_version,
         manage_config => false,
         before        => Augeas["jenkins::augeas: ${name}"],
       }
     }
-    false,'false': { # lint:ignore:quoted_booleans
-      #do nothing
+    false: {
```

Comment:
I am not sure why you want to keep the false case. As an outsider to the code, it looks idiotic ;-)